[jaeger] Major refactoring of templates to simplify

jkowall opened the pull request:
# [jaeger] Major refactoring for Jaeger v2

## Summary

This PR modernizes the Jaeger Helm chart for **Jaeger v2**, which is built on the OpenTelemetry Collector framework. It removes legacy v1 components and consolidates to a single deployment model with simplified, more maintainable templates.

## Breaking Changes ⚠️

### Values Schema Changes
- **`allInOne:` → `jaeger:`** - The main deployment section has been renamed
- **`schema:` section removed** - Cassandra schema job no longer needed (Jaeger v2 handles schema internally)
- **`hotrod:` section removed** - Demo app removed from chart
- **Storage configuration** moved to `config.extensions.jaeger_storage` using native OTel Collector syntax

### Migration Guide

```yaml
# Before (v1)
allInOne:
  image:
    tag: "1.x.x"

# After (v2)
jaeger:
  image:
    tag: "2.13.0"
```

For storage configuration, see the updated README.md examples.

## Changes

### Templates Removed (Legacy v1 Components)
- `allinone-agent-svc.yaml`, `allinone-collector-svc.yaml`, `allinone-query-svc.yaml` - Separate services
- `cassandra-schema-job.yaml` & `cassandra-schema-sa.yaml` - Schema initialization
- `elasticsearch-secret.yaml` - ES secret management
- `es-index-cleaner-cronjob.yaml`, `es-rollover-cronjob.yaml`, `es-lookback-cronjob.yaml` & their SA files - Now in es-maintenance.yaml
- `hotrod-*.yaml` - Demo application
- `spark-sa.yaml` - Merged into spark-dependencies.yaml

### Templates Renamed
- allinone-deploy.yaml → jaeger-deploy.yaml
- allinone-sa.yaml → `jaeger-sa.yaml`
- allinone-ing.yaml → `jaeger-ing.yaml`
- allinone-configmap.yaml → `jaeger-configmap.yaml`
- `spark-cronjob.yaml` → `spark-dependencies.yaml`

### Templates Added/Consolidated
- service.yaml - Unified service exposing all ports (agent, collector, query, OTLP)
- `es-maintenance.yaml` - Consolidated ES maintenance jobs with their ServiceAccounts

### Helper Functions Cleaned Up (_helpers.tpl)
Removed ~100 lines of unused helpers including schema, hotrod, and legacy ES helpers.

### values.yaml Improvements
- Clear section headers distinguishing `storage.*` (connection settings) from top-level `cassandra:`/`elasticsearch:` (subchart provisioning)
- Removed deprecated/unused settings
- Added documentation comments throughout

### Documentation
- Completely rewritten README.md for Jaeger v2
- Added "Overriding the Jaeger Version" section
- Added Spark Dependencies README link
- Clarified Cassandra is external-only

### Bug Fixes
- Fixed duplicate annotations block in service.yaml
- Fixed storage-configmap.yaml to render `Values.config` directly

## Stats
- **32 files changed**
- **+700 insertions, -1,696 deletions** (net ~1,000 lines removed)

## Testing
- [x] `helm lint charts/jaeger` passes
- [x] `make test` passes (deploys to local k8s)
- [x] Templates render correctly for memory, elasticsearch, and cassandra storage types

## Checklist
- [x] Chart version bumped (4.1.5 → 4.2.0)
- [x] README.md updated
- [x] values.yaml documented
- [x] .gitignore updated (added .vscode/)
- [x] All commits signed off (DCO)

jkowall committed:
Major refactoring of templates to simplify

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall committed:
Fixing CI issues

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall committed:
Fixing CI issues, verified locally first

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall commented:
There were still CI issues, but this push should fix it

## CI Fix - Restore Provisioned Elasticsearch Logic & Stability
- **Goal**: Fix CI failure where Jaeger failed to connect to the provisioned Elasticsearch instance.
- **Problem 1**: Removing the bespoke DSL inadvertently removed the logic that wired up `provisionDataStore.elasticsearch=true` in [_helpers.tpl](cci:7://file:///home/jkowall/helm-charts-F/charts/jaeger/templates/_helpers.tpl:0:0-0:0).
- **Fix 1**:
  - Re-introduced `elasticsearch.env` helper in [_helpers.tpl](cci:7://file:///home/jkowall/helm-charts-F/charts/jaeger/templates/_helpers.tpl:0:0-0:0) to auto-configure `ES_SERVER_URLS`.
  - Updated `storage.env` and [es-maintenance.yaml](cci:7://file:///home/jkowall/helm-charts-F/charts/jaeger/templates/es-maintenance.yaml:0:0-0:0) to use this logic.
- **Problem 2**: CI Timeout and Configuration Errors.
  - [values.yaml](cci:7://file:///home/jkowall/helm-charts-F/charts/jaeger/values.yaml:0:0-0:0) contained an invalid `enabled: false` field in `volumeClaimTemplate` causing K8s warnings.
  - The CI environment was failing the Elasticsearch Readiness Probe due to short timeouts (`1s`) and slow runners.
  - Jaeger v2 (used in `all-in-one`) failed to start with `empty backend configuration` error because [storage-configmap.yaml](cci:7://file:///home/jkowall/helm-charts-F/charts/jaeger/templates/storage-configmap.yaml:0:0-0:0) was generating empty blocks for `primary_store` and `archive_store`.
- **Fix 2**:
  - Removed invalid configuration from [values.yaml](cci:7://file:///home/jkowall/helm-charts-F/charts/jaeger/values.yaml:0:0-0:0).
  - Updated [Makefile](cci:7://file:///home/jkowall/helm-charts-F/Makefile:0:0-0:0) to increase `timeout` to `10s` and increase `initialDelaySeconds` and `failureThreshold` for Elasticsearch readiness probes.
  - Updated [storage-configmap.yaml](cci:7://file:///home/jkowall/helm-charts-F/charts/jaeger/templates/storage-configmap.yaml:0:0-0:0) to fully restore the automatic configuration of `primary_store` and `archive_store` when `provisionDataStore.elasticsearch` is enabled.
  - **Critical Fix**: Modified [storage-configmap.yaml](cci:7://file:///home/jkowall/helm-charts-F/charts/jaeger/templates/storage-configmap.yaml:0:0-0:0) to embed credentials directly in `server_urls` (e.g., `https://user:pass@host:port`) because Jaeger v2 configuration rejection of `username`/`password` keys and failed to respect environment variable overrides when a config file was present. This ensures successful connection and authentication.

jkowall committed:
Fixing linting issue

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall committed:
Fixing linting issue #2

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

yurishkuro commented on charts/jaeger/templates/cassandra-schema.yaml:
```diff
@@ -58,6 +58,7 @@ spec:
         {{- end }}
         resources:
           {{- toYaml .Values.schema.resources | nindent 10 }}
+        {{- if or .Values.schema.extraConfigmapMounts .Values.storage.cassandra.tls.enabled }}
```
The main binary has the ability to create schema, we don't need an external job anymore. Can we drop this template?

yurishkuro commented on charts/jaeger/README.md:
```diff
@@ -1,440 +1,112 @@
-  image:
-    tag: "2.13.0"
+  extraEnv:
+    - name: SPAN_STORAGE_TYPE
```
I'm confused - how do these env vars get applied? jaeger-v1 used to understand them via viper, but with OTEL config they are no longer respected

yurishkuro commented on charts/jaeger/values.yaml:
```diff
@@ -199,40 +199,9 @@ storage:
-      ca: /es-tls/ca-cert.pem
+    # Deprecated: use extraEnv
+    # indexPrefix: jaeger
+    # serverUrls: http://elasticsearch:9200
```
in configmap we have things like these:
```
            {{ if or .Values.provisionDataStore.elasticsearch (eq .Values.storage.type "elasticsearch") }}
            {{- $es := .Values.storage.elasticsearch | default dict -}}
            {{- $scheme := $es.scheme | default "http" -}}
            {{- $port := $es.port | default 9200 -}}
            {{- $user := $es.user | default "elastic" -}}
            {{- $password := .Values.elasticsearch.secret.password | default "changeme" -}}
```
doesn't it mean they are trying to read values from this section?

yurishkuro commented on charts/jaeger/README.md:
````diff
@@ -1,440 +1,112 @@
+```yaml
+storage:
+  type: cassandra
+  cassandra:
````
this seems inconsistent with elasticsearch, should we not be using identical configuration mechanisms?

jkowall committed:
Address PR #702 review feedback

- Version: Changed from 5.0.0 to 4.2.0 per reviewer request
- Added experimental/unstable warning to README and NOTES.txt
- Removed cassandra-schema.yaml (Jaeger v2 handles schema internally)
- Removed schema: section from values.yaml
- Refactored storage-configmap.yaml to use blob-style config
- Updated README with native Jaeger/OTEL config examples
- Clarified Cassandra support status in documentation

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall committed:
CI fixes

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall committed:
remove duplicate args and support external ES env vars

- Remove duplicate arguments for:
  - `jaeger-es-index-cleaner` (`numberOfDays`)
  - `jaeger-es-rollover` (`rollover`)
  - `jaeger-es-lookback` (`lookback`)
- Update `elasticsearch.env` helper to generate environment variables when `storage.type` is `elasticsearch`, enabling support for external Elasticsearch clusters.

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall committed:
Address PR #702 review feedback

- Remove .vscode/ folder and add to .gitignore
- Add 'Overriding the Jaeger Version' documentation section
- Fix Cassandra wording: clarify chart doesn't provision Cassandra
- Add link to Spark Dependencies README for env vars
- Simplify storage-configmap.yaml condition per review feedback

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall committed:
Simplify storage-configmap.yaml: render Values.config directly

Per review feedback, removed parameterization helpers and now
render the entire .Values.config section as YAML directly instead
of cherry-picking subsections. This is simpler and more maintainable.

Removed unused helpers:
- jaeger.extensionsConfig
- jaeger.receiversConfig
- jaeger.processorsConfig
- jaeger.exportersConfig
- jaeger.serviceExtensions
- jaeger.serviceReceivers
- jaeger.serviceProcessors
- jaeger.serviceExporters

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall committed:
fix: remove duplicate annotations block in service.yaml

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall committed:
docs: clarify storage vs subchart sections in values.yaml

Added clear section headers and comments to distinguish:
- storage.* - connection settings for external storage backends
- cassandra/elasticsearch (top-level) - subchart values for provisioning

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall commented on charts/jaeger/values.yaml:
```diff
@@ -168,9 +167,6 @@ storage:
   cassandra:
```
Added comments

jkowall committed:
docs: clarify storage.elasticsearch settings in values.yaml

Explain that ES config is now in config.extensions.jaeger_storage
and that storage.elasticsearch is only used by ES maintenance jobs.

Signed-off-by: Jonah Kowall <jkowall@kowall.net>

jkowall commented on charts/jaeger/values.yaml:
```diff
@@ -192,47 +187,15 @@ storage:
-      subPath: ca-cert.pem
-      # How ES_TLS_CA variable will be set in the various components
-      ca: /es-tls/ca-cert.pem
+    # Deprecated: use extraEnv
```
The comment now clearly explains:

Jaeger v2 ES config goes in config.extensions.jaeger_storage (at the top of values.yaml)
storage.elasticsearch is only used by the ES maintenance jobs (esIndexCleaner, esRollover, esLookback) via the elasticsearch.env helper template
The old commented-out settings (indexPrefix, serverUrls) were removed since they're not used anymore - they were just confusing leftovers from v1.

jkowall commented on charts/jaeger/values.yaml:
Yes, I would do it, but here are some things to consider:

**Pros of renaming to `jaeger:`**
- Cleaner and more intuitive
- Aligns with v2 architecture (there's only one deployment mode now)
- `jaeger.image`, `jaeger.service`, `jaeger.ingress` reads better

**Cons:**
- Breaking change for anyone who has `allInOne:` in their existing values overrides
- Requires updating all templates that reference `.Values.allInOne`

Since this PR is already a major breaking change (v1 → v2 migration), now would be the right time to make this rename if you want to do it. Users upgrading will need to review their values anyway.

Want me to make this change? It would touch:
- values.yaml - rename the section
- All templates referencing `.Values.allInOne` 
- README.md - update examples

jkowall committed:
refactor: rename allInOne to jaeger in values and templates

BREAKING CHANGE: The 'allInOne' values section is now 'jaeger'.

Migration: Replace 'allInOne:' with 'jaeger:' in your values files.

- Renamed allInOne section to jaeger in values.yaml
- Renamed template files from allinone-* to jaeger-*
- Updated all template references
- Updated README.md examples

Signed-off-by: Jonah Kowall <jkowall@kowall.net>
